src/account/account.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from "@nestjs/common";
+import { utils } from "ethers";
+import { getRPCProvider } from "../utils/client";
+import { Currency } from "../models/Currency";
+import { Errors } from "../models/Errors";
+import { PartialBlockIdentifier } from "../models/PartialBlockIdentifier";
+
+@Injectable()
+export class AccountService {
+
+  public async getBalance(accountAddress: string, blockNumber?: number) {
+    const provider = getRPCProvider()
+    let blockIdentifier = new PartialBlockIdentifier()
+    let balances = []
+    
+    if (blockNumber){      
+      // historical balance lookup            
+      const block = await provider.getBlock(blockNumber)      
+      
+      if (!block || (block.number !== blockNumber)) {        
+        return Errors.BLOCK_NOT_FOUND;  
+      }
```

Comment:
Maybe better to move this to PartialBlockIdentifier `validate` function?

Reply:
Agree, moved